hpx/parallel/segmented_algorithms/transform.hpp

```diff
@@ -0,0 +1,876 @@
+                    {
+                        // handle any remote exceptions, will throw on error
+                        std::list<std::exception_ptr> errors;
+                        parallel::util::detail::handle_remote_exceptions<
+                            ExPolicy
+                        >::call(r, errors);
+                        auto rl = r.back().get();
+                        auto olast1 = traits1::compose(send1, hpx::util::get<0>(rl));
+                        auto olast2 = traits2::compose(send2, hpx::util::get<1>(rl));
+                        auto odest = traits3::compose(sdest, hpx::util::get<2>(rl));
+                        return hpx::util::make_tuple(olast1, olast2, odest);
+                    },
+                    std::move(segments))));
+        }
+
+        ///////////////////////////////////////////////////////////////////////
+        // segmented implementation
+        template <
+            typename ExPolicy, typename InIter1, typename InIter2,
+            typename OutIter, typename F, typename Proj1, typename Proj2>
+        typename util::detail::algorithm_result<
+            ExPolicy,
+            hpx::util::tagged_tuple<
+                tag::in1(InIter1), tag::in2(InIter2), tag::out(OutIter)
+            >
+        >::type
+        transform_(ExPolicy && policy,
+            InIter1 first1, InIter1 last1, InIter2 first2, OutIter dest, F && f,
+            Proj1 && proj1, Proj2 && proj2, std::true_type)
+        {
+            typedef parallel::execution::is_sequenced_execution_policy<
+                    ExPolicy
+                > is_seq;
+            typedef util::detail::algorithm_result<ExPolicy,
+            hpx::util::tagged_tuple<
+                tag::in1(InIter1), tag::in2(InIter2), tag::out(OutIter)>
+            > result;
+            auto last2 = first2;
+            detail::advance(last2,std::size_t(last1 - first1));
+            if (first1 == last1)
+            {
+                return result::get(hpx::util::make_tuple<InIter1, InIter2,
+                    OutIter>(std::move(last1), std::move(last2),
+                    std::move(dest)));
+            }
+
+            typedef hpx::traits::segmented_iterator_traits<InIter1>
+                iterator1_traits;
+            typedef hpx::traits::segmented_iterator_traits<InIter2>
+                iterator2_traits;
+            typedef hpx::traits::segmented_iterator_traits<OutIter>
+                iterator3_traits;
+            return hpx::util::make_tagged_tuple<tag::in1, tag::in2, tag::out>(
+                segmented_transform(transform_binary<hpx::util::tuple<
+                typename iterator1_traits::local_iterator,
+                typename iterator2_traits::local_iterator,
+                typename iterator3_traits::local_iterator> >(),
+                std::forward<ExPolicy>(policy), first1, last1, first2,
+                dest, std::forward<F>(f), proj1, proj2, is_seq()));
```

Comment:
`proj1` and `proj2` should be perfectly forwarded. Please fix formatting (see above).